### ISSUE TITLE
Issues with loading the ruleset

### ISSUE THREAD

**Issue by pjungermann** [dweber019/backstage-plugin-api-docs-spectral-linter#56]
TL;DR: Currently, I assume that the issue is caused by change at `stoplightio/spectral`, not 100% sure, though.

I used the following inside `app-config.yaml`

```yaml
spectralLinter:
  asyncApiRulesetUrl: spectral:asyncapi
  openApiRulesetUrl: spectral:oas
```

Internally, it gets translated to a `.spectral.yaml` file like

```yaml
extends:
  - spectral:oas
```

It used to work. However, when I use this inside of Backstage, it fails with "Provided ruleset is not an object". I've tried this with the current master of backstage/backstage as well:
![Screenshot 2023-08-18 at 11 23 11](https://github.com/dweber019/backstage-plugin-api-docs-spectral-linter/assets/2513586/3ede0c5d-2519-41b8-ba56-43d1c3087aad)

It fails at
```js
function assertValidRuleset(ruleset, format = 'js') {
    var _a;
    if (!(0, json_1.isPlainObject)(ruleset)) {
        throw new errors_1.RulesetValidationError('invalid-ruleset-definition', 'Provided ruleset is not an object', []);
    }
    // ...
}
```

and `ruleset` is an instance of the class `Ruleset`.

and `json_1` comes from `@spotlight/json`

```js
function isPlainObject(e) {
  if ("object" != typeof e || null === e) return !1;
  const t = Object.getPrototypeOf(e);
  return null === t || t === Object.prototype || "function" == typeof e.constructor && Function.toString.call(Object) === Function.toString.call(e.constructor)
}
```

The issue might be at
```js
    setRuleset(ruleset) {
        this.ruleset = ruleset instanceof ruleset_1.Ruleset ? ruleset : new ruleset_1.Ruleset(ruleset);
    }
```
where `ruleset` actually is an instance of a class Ruleset, however, the `instanceof` check fails.

The ruleset instance itself was created by
```ts
import { Spectral } from '@stoplight/spectral-core';
import { bundleAndLoadRuleset } from '@stoplight/spectral-ruleset-bundler/with-loader';
import { fetch } from '@stoplight/spectral-runtime';

// ...

    const spectral = new Spectral();
    const ruleSet = await bundleAndLoadRuleset("/.spectral.yaml", {
      fs,
      fetch
    });
    spectral.setRuleset(ruleSet);
```

Currently, the versions are fixed to specific patch versions. For spectral-core, the latest version is 1.18.3.

Maybe, it can be resolved by upgrading (e.g., PR #25 which is not at the recent version due to "Dependabot updates are paused").

**Comment by dweber019** [comment]
Thanks for the input. Will have a look at this tomorrow.

**Comment by dweber019** [comment]
Fix with version 0.2.5.
